src/xeus_calc_interpreter.cpp

```diff
@@ -46,87 +46,57 @@ namespace xeus_calc
         return spaced_expr;
     }
 
-    std::string parse_rpn(const std::string& infix, publish_type publish)
+    std::string parse_rpn(std::string expression)
     {
-        const std::string ops = "-+/*^";
-        std::stringstream parsed_infix;
-        std::size_t parenthesis_counter = 0; // used for error management purpose
-
-        std::stack<int> operators;
-
-        std::stringstream input(infix);
+        std::istringstream infix(expression);
         std::string token;
-        while (std::getline(input, token, ' '))
+        std::string output;
+        std::string operators = "+-*/^";
```

Comment:
Like the `compute_rpn` function, you should use a map instead for more flexibility if you want to extend the supported functions.